component/module/git/git.go

```diff
@@ -220,8 +227,13 @@ func (c *Component) Update(args component.Arguments) (err error) {
 // controller. pollFile must only be called with c.mut held.
 func (c *Component) pollFile(ctx context.Context, args Arguments) error {
 	// Make sure our repo is up-to-date.
+	var updateVcsErr *vcs.UpdateFailedError
 	if err := c.repo.Update(ctx); err != nil {
-		return err
+		if errors.As(err, &updateVcsErr) {
+			level.Error(c.log).Log("msg", "failed to update repository", "err", err)
+		} else {
+			return err
+		}
```

Comment:
I'm worried this fix is too broad, and changes behavior we don't want to change: this (and the other change) is now hiding the error from the user even if the repo has never been cloned before.

Rather than hiding the error, I'd like to see the behavior changed to the following:

* Always return an error if `git clone` fails, since `git clone` is only used when the cache does not exist. 
* If a `git pull` fails while building the component via New, do not return an error via New, but still make sure the component's health gets marked as unhealthy. 
* In all other cases, if `git pull` fails, return an error. 

These changes will fix #5708 while never hiding the failure from the component health or the Flow controller.

Reply:
Correct me if I'm wrong but I assume my changes would only hide `git pull` error from being returned and for other errors, they'll be returned as normal. `git clone` error is always returned due to its criticality for the component and my changes assumedly doesn't change this behavior

You're right on hiding `git pull` error from Flow controller isn't ideal and I should mark it the component unhealthy whenever a failed `git pull` error occurs. Let me know if this resolves your concern 👀

Reply:
> Correct me if I'm wrong but I assume my changes would only hide `git pull` error from being returned and for other errors, they'll be returned as normal. 

You're right, sorry for the misunderstanding here; I didn't catch that there was two different error types depending on the operation that failed.